airflow/www/views.py

```diff
@@ -1725,14 +1725,38 @@ def index(self):
         qry = None
         # filter the dags if filter_by_owner and current user is not superuser
         do_filter = FILTER_BY_OWNER and (not current_user.is_superuser())
+        owner_mode = conf.get('webserver', 'OWNER_MODE').strip().lower()
+
+        if do_filter and owner_mode not in ['user', 'ldapgroup']:
+            logout_user()
+            flash("Configuration error: "
```

Comment:
Would adding an 'elif' in ConfigParserWithDefaults._validate be sufficient or do you know if a better config validation will happen in the near future?